sys/include/event/thread.h

```diff
@@ -12,6 +12,39 @@
  * @ingroup     sys_event
  * @brief       Provides utility functions for event handler threads
  *
+ * Usage
+ * =====
+ *
+ * By using the module `event_thread`, the event queues
+ * @ref EVENT_PRIO_HIGHEST, @ref EVENT_PRIO_MEDIUM, and
+ * @ref EVENT_PRIO_LOWEST are provided and declared in the header
+ * `event/thread.h`. With default settings, one or more threads to handle these
+ * queues are automatically started during startup due to the module
+ * `auto_init`.
+ *
+ * By default, only a single thread with priority `EVENT_THREAD_MEDIUM_PRIO` is
+ * started that handles all three event queues according to their priority.
+ * An already started event handler will not be preempted in this case by an
+ * incoming higher priority event. Still, lower priority event queues will only
+ * be worked on once the higher priority queues are empty. Hence, the worst case
+ * latency is increased by the worst case duration of any possible lower
+ * priority event in this configuration.
+ *
+ * By using module `event_thread_highest`, the highest priority queue gets its
+ * own thread. With this, events of the highest priority can preempt already
+ * running event handlers of medium and lowest priority.
+ *
+ * By using module `event_thread_medium`, the lowest priority events are handled
+ * in their own thread. With this, events of at least medium priority can
+ * preempt already running events of the lowest priority.
+ *
+ * By using both module `event_thread_highest` and `event_thread_medium`, each
+ * event queue gets its own thread. So higher priority events will always
+ * preempt events of lower priority in this case.
```

Comment:
If these modules do not have dependencies they could be `DEFAULT_MODULES` no?

Reply:
I'd say the default should use only one thread to conserve resources.
Only enable additional threads when needed.